lib/vfscore/main.c

```diff
@@ -329,24 +348,44 @@ UK_TRACEPOINT(trace_vfs_pread, "%d %p 0x%x 0x%x", int, void*, size_t, off_t);
 UK_TRACEPOINT(trace_vfs_pread_ret, "0x%x", ssize_t);
 UK_TRACEPOINT(trace_vfs_pread_err, "%d", int);
 
-ssize_t pread(int fd, void *buf, size_t count, off_t offset)
+/*
+ * Some libc's define some macros that remove the 64 suffix
+ * from some system call function names (e.g., <unistd.h>, <fcntl.h>).
+ * We need to undefine them here so that our system call
+ * registration does not fail in such a case.
+ */
+#ifdef pread64
+#undef pread64
+#endif
+
+UK_LLSYSCALL_R_DEFINE(ssize_t, pread64, int, fd,
+		      void *, buf, size_t, count, off_t, offset)
 {
 	trace_vfs_pread(fd, buf, count, offset);
 	struct iovec iov = {
 			.iov_base	= buf,
 			.iov_len	= count,
 	};
-	int bytes;
+	ssize_t bytes;
 
-	bytes = preadv(fd, &iov, 1, offset);
+	bytes = uk_syscall_r_preadv((long) fd, (long) &iov,
+				    (long) 1, (long) offset);
 	if (bytes < 0)
 		trace_vfs_pread_err(bytes);
 	else
 		trace_vfs_pread_ret(bytes);
 	return bytes;
 }
 
+#if UK_LIBC_SYSCALLS
+ssize_t pread(int fd, void *buf, size_t count, off_t offset)
```

Comment:
This does bring the libc-style wrapper for `pread` but causes an undefined reference if the libc-style wrapper of `pread64` were to be used.

Reply:
Hum... I was actually expecting that the libc is mapping `pread64()` to `pread()` with a macro, so the other libc wrapper would not be needed. In fact, on an libc implementation that does not do this the symbol would be missing. I guess it does not hurt if I provide both symbols: The libc implementation makes the decision if we link to both or just one of them and the other symbol would be unused and could be removed.

Reply:
Hey @mogasergiu, I enabled the 64 variants of the system call symbols with the already available `LFS64()` macro. It is a separate commit on the updated PR. Thanks, Simon